tests/MongoDB.Driver.Tests/Linq/MongoQueryableTests.cs

```diff
@@ -1706,6 +1709,37 @@ public void Where_method_with_predicated_any()
                 "{ $match : { 'G' : { '$elemMatch' : { 'D' : \"Don't\" } } } }");
         }
 
+        [SkippableFact]
+        public void AsQueryable_in_transaction()
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded).Supports(Feature.Transactions);
+            if (CoreTestConfiguration.Cluster.Description.Type == ClusterType.Sharded)
+            {
+                RequireServer.Check().Supports(Feature.ShardedTransactions);
+            }
+
+            using (var session = DriverTestConfiguration.Client.StartSession())
+            {
+                session.StartTransaction();
+                try
+                {
+                    __collection.InsertOne(session, new Root());
+
+                    var result_not_in_transaction = CreateQuery(null).Count(); // checks AsQueryable with null session (outside transaction)
+
+                    result_not_in_transaction.Should().Be(2);
```

Comment:
Took me a while to figure out why there was data in this collection already, but I got it in the end :)